src/main/java/com/iexec/core/task/TaskService.java

```diff
@@ -354,6 +357,18 @@ private void uploadRequested2UploadingResult(Task task) {
         }
     }
 
+    private void uploadRequested2UploadRequestTimeout(Task task) {
```

Comment:
actually, that's possible only for replicate detectors because we can update replicaStatus from the outside. For tasks, we don't do any logic inside detectors because we need to tell the `taskExecutorEngine` to do that for us. So if a timeout is detected we request a `trytoUpgradeTaskStatus()` which would handle everything.

Reply:
ah you're right, somehow I thought we modify directly the status of Task in the detector, my bad.

Reply:
No problem 👍